typhos/cli.py

```diff
@@ -534,34 +574,42 @@ def typhos_run(
             scroll_option=scroll_option,
             show_displays=show_displays,
         )
-    if suite:
-        if initial_size is not None:
-            try:
-                initial_size = QtCore.QSize(
-                    *(int(opt) for opt in initial_size.split(','))
-                )
-            except TypeError as exc:
-                raise ValueError(
-                    "Invalid --size argument. Expected a two-element pair "
-                    "of comma-separated integers, e.g. --size 1000,1000"
-                ) from exc
-
-        def exit_early():
-            logger.warning(
-                "Exiting typhos early due to --exit-after=%s CLI argument.",
-                exit_after
+
+    if suite is None:
+        logger.debug("Suite creation failure")
+        return None
+
+    if initial_size is not None:
+        try:
+            initial_size = QtCore.QSize(
+                *(int(opt) for opt in initial_size.split(','))
             )
-            sys.exit(0)
+        except TypeError as exc:
+            raise ValueError(
+                "Invalid --size argument. Expected a two-element pair "
+                "of comma-separated integers, e.g. --size 1000,1000"
+            ) from exc
+
+    def exit_early():
+        logger.warning(
+            "Exiting typhos early due to --exit-after=%s CLI argument.",
+            exit_after
+        )
+
+        if screenshot_filename is not None:
+            suite.save_device_screenshots(screenshot_filename)
+
+        sys.exit(0)
 
-        if exit_after is not None and exit_after >= 0:
-            QtCore.QTimer.singleShot(exit_after * 1000.0, exit_early)
+    if exit_after is not None and exit_after >= 0:
+        QtCore.QTimer.singleShot(exit_after * 1000.0, exit_early)
 
-        return launch_suite(suite, initial_size=initial_size)
+    return launch_suite(suite, initial_size=initial_size)
 
 
 def typhos_cli(args):
     """Command Line Application for Typhos."""
-    args = parser.parse_args(args)
+    args = typing.cast(TyphosArguments, parser.parse_args(args))
```

Comment:
There is also a built-in feature of argparse to fill a namespace instead of creating one. I'm not sure which way is better from a maintainability and type-checking standpoint.
```
args = parser.parse_args(args, TyphosArguments())
```

Reply:
Ah, thanks for that tip! I think your way is better than the type cast :+1: